module-2/backend/week_three.md

```diff
@@ -10,14 +10,55 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+
+* rails new example_project
 2. What do Models generally inherit from in rails?
+
+* ApplicationRecord
 3. What do Controllers generally inherit from in a rails project?
+
+* ApplicationController
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+* resources :horses, only: [:show]
 5. What rake task is useful when looking at routes, and what information does it give you?
+
+* rake routes. This tells us the pathing we have access to to take advantage of in our controllers
 6. What is an example of a route helper? When would you use them?
+
+* resources :horses is a route helper. We generally will be using these for all production apps. 
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+
+* Off the top of my head, I don't know. 
 8. What are strong params and why are the necessary?
+
+* Strong params help us make our app more secure by limiting what the user has access to. I can permit what I want them to have. And having strong params is useful in my create method on my controller. 
 9. What role does `form_for` play in helping us create our forms?
+
+* form_for is a rails helper method that allows us access to many methods that can be used to make up our form in our block. 
 10. How does `form_for` know where to submit the user's input?
+
+* I think based on the action that we offer. I'd have to look it up. 
```

Comment:
It uses the argument you pass to it. In the example below, the argument is `@horse`, which is an instance of Horse. It checks to see if it's been saved to the database (e.g. is it an actual horse with a name, and id, etc., or just an instance you created with `Horse.new`), and then uses that to make a path to either post to `/horses` (for create when it hasn't been persisted), or patch to `/horses/1` (for update when it has).